engine/src/main/scala/cromwell/engine/workflow/tokens/TokenQueue.scala

```diff
@@ -57,32 +65,48 @@ final case class TokenQueue(queues: Map[String, Queue[TokenQueuePlaceholder]],
       val remainingHogGroups = queuesRemaining.tail
       val leaseTry = pool.tryAcquire(hogGroup)
 
-      leaseTry match {
-        case thl: TokenHoggingLease =>
-          val oldQueue = queues(hogGroup)
-          val (placeholder, newQueue) = oldQueue.dequeue
-          val (newQueues, newQueueOrder) = if (newQueue.isEmpty) {
-            (queues - hogGroup, remainingHogGroups ++ queuesTried)
-          } else {
-            (queues + (hogGroup -> newQueue), remainingHogGroups ++ queuesTried :+ hogGroup)
-          }
-          DequeueResult(Some(LeasedActor(placeholder, thl)), TokenQueue(newQueues, newQueueOrder, eventLogger, pool))
-        case TokenTypeExhausted =>
-          // The pool is completely full right now, so there's no benefit trying the other hog groups:
-          eventLogger.outOfTokens(tokenType.backend)
-          DequeueResult(None, this)
-        case HogLimitExceeded =>
-          eventLogger.flagTokenHog(hogGroup)
-          recursingDequeue(queues, queuesTried :+ hogGroup, remainingHogGroups)
+      val oldQueue = queues(hogGroup)
+
+      if (oldQueue.isEmpty) {
+        // We should have caught this above. But just in case:
+        logger.warn(s"Programmer error: Empty token queue value still present in TokenQueue: $hogGroup *and* made it through into recursiveDequeue(!): $hogGroup")
+        recursingDequeue(queues, queuesTried :+ hogGroup, remainingHogGroups)
+      } else {
+        leaseTry match {
+          case thl: TokenHoggingLease =>
+            val (placeholder, newQueue) = oldQueue.dequeue
+            val (newQueues, newQueueOrder) = if (newQueue.isEmpty) {
+              (queues - hogGroup, remainingHogGroups ++ queuesTried)
+            } else {
+              (queues + (hogGroup -> newQueue), remainingHogGroups ++ queuesTried :+ hogGroup)
+            }
+            DequeueResult(Some(LeasedActor(placeholder, thl)), TokenQueue(newQueues, newQueueOrder, eventLogger, pool))
```

Comment:
```suggestion
            DequeueResult(Option(LeasedActor(placeholder, thl)), TokenQueue(newQueues, newQueueOrder, eventLogger, pool))
```

Reply:
OK but I still think this is strictly worse than using `Some` in these situations